wdl2cwl/main.py

```diff
@@ -605,24 +627,75 @@ def get_expr_apply(self, wdl_apply_expr: WDL.Expr.Apply) -> str:
         elif function_name == "glob":
             only_arg = arguments[0]
             return self.get_expr(only_arg)
+        elif function_name == "select_first":
+            array_obj = arguments[0]
+            array_items = [self.get_expr(item) for item in array_obj.items]  # type: ignore
+            items_str = ", ".join(array_items)
+            return f"[{items_str}].find(element => element !== null) "
+        elif function_name == "_mul":
+            left_operand, right_operand = arguments
+            left_str = self.get_expr(left_operand)
+            right_str = self.get_expr(right_operand)
+            return f"{left_str}*{right_str}"
+        elif function_name == "_eqeq":
+            left_operand, right_operand = arguments
+            left_str = self.get_expr(left_operand)
+            right_str = self.get_expr(right_operand)
+            return f"{left_str} === {right_str}"
+        elif function_name == "ceil":
+            only_arg = self.get_expr(arguments[0])  # type: ignore
+            return f"Math.ceil({only_arg}) "
+        elif function_name == "_div":
+            left_operand, right_operand = arguments
+            left_str = self.get_expr(left_operand)
+            right_str = self.get_expr(right_operand)
+            return f"{left_str}/{right_str}"
+        elif function_name == "size":
+            left_operand, right_operand = arguments
+            left_str = self.get_expr(left_operand)
+            size_unit = self.get_expr(right_operand)[1:-1]
+            with WDLSourceLine(size_unit, ConversionException):
+                if size_unit == "Ki" or size_unit == "K":
+                    unit_value = "1024"
+                elif size_unit == "Mi" or size_unit == "M":
+                    unit_value = "1024^2"
+                if size_unit == "Gi" or size_unit == "G":
+                    unit_value = "1024^3"
+                elif size_unit == "Ti" or size_unit == "T":
+                    unit_value = "1024^4"
+                elif size_unit == "Pi" or size_unit == "P":
+                    unit_value = "1024^5"
+                else:
+                    raise WDLSourceLine(size_unit, ConversionException).makeError(
+                        f"Size Unit '{size_unit}' not yet handled."
+                    )
+            return (
+                "(function(size_of=0)"
+                + "{"
+                + f"{left_str}.forEach(function(element)"
+                + "{"
+                + "size_of += element.size"
+                + "})}"
+                + f") / {unit_value}"
```

Comment:
What if `size()` is called on a single `File` (no array)?

Reply:
Should I add a condition to be sure it's an Array? I actually don't know what a File instance with the size function looks like to I can't really say.

Reply:
> I actually don't know what a File instance with the size function looks like to I can't really say.

I don't either, but you could make a variation of `bwa.wdl` that calls `size` on a single file to find out..